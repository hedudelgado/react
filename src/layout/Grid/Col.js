import React from 'react';
import {
  node, bool, string, oneOf,
} from 'prop-types';
import classNames from 'classnames';

import BaseStyle from '../BaseStyle';

const SIZES = ['auto', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'];

const Col = ({
  children,
  border,
  sm,
  md,
  lg,
  size,
  className: classNameProps,
  ...otherProps
}) => {
  const baseSizeClass = () => (sm && 'sm-') || (md && 'md-') || (lg && 'lg-') || '';
  const classes = classNames(
    classNameProps,
    [`${baseSizeClass()}col`],
    {
      border,
      col: size,
      [`sm-col-${sm}`]: sm,
      [`md-col-${md}`]: md,
      [`lg-col-${lg}`]: lg,
      [`col-${size}`]: size,
    },
  );
  return <BaseStyle className={classes} {...otherProps}>{children}</BaseStyle>;
};

Col.propTypes = {
  /**
   * Any Node
   */
  children: node.isRequired,
  /**
   * optional: displays column borders
   */
  border: bool,
  /**
   * Size for a static column
   * Use for non-responsive Grid only
   */
  size: oneOf(SIZES),
  /**
   * tablet breakpoint
   */
  sm: oneOf(SIZES),
  /**
   * tabletLarge breakpoint
   */
  md: oneOf(SIZES),
  /**
   * desktop breakpoint
   */
  lg: oneOf(SIZES),
  /**
   * additional css classes
   */
  className: string,
};

Col.defaultProps = {
  border: false,
};

export default Col;
