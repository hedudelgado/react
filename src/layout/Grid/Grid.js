import React from 'react';
import { node, bool } from 'prop-types';
import classNames from 'classnames';

import BaseStyle from '../BaseStyle';

const Grid = ({
  children,
  border,
  ...otherProps
}) => {
  const classes = classNames(
    'clearfix',
    {
      border,
    },
  );
  return <BaseStyle className={classes} {...otherProps}>{children}</BaseStyle>;
};

Grid.propTypes = {
  /**
   * Any Node
   */
  children: node.isRequired,
  /**
   * optional: displays column borders
   */
  border: bool,
};

export default Grid;
