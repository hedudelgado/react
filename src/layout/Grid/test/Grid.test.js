import React from 'react';
import { shallow, mount } from 'enzyme';

import Grid from '../Grid';
import Col from '../Col';

describe('Grid Component', () => {
  describe('Main Container', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Grid><Col>test</Col></Grid>);
    });
    it('should render the component', () => {
      expect(wrapper).not.toBeEmptyRender();
      expect(wrapper).toMatchSnapshot();
    });
    it('should render a container', () => {
      expect(wrapper).toHaveClassName('clearfix');
    });
    it('should render borders', () => {
      wrapper = shallow(<Grid border>test</Grid>);
      expect(wrapper).toHaveClassName('border');
    });
  });
  describe('Columns', () => {
    const colSizes = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'];

    describe('default behaviour', () => {
      let wrapper;
      beforeEach(() => {
        wrapper = shallow(<Col size="6" key="0">col1</Col>);
      });
      it('should show 1 column', () => {
        expect(wrapper).toContainExactlyOneMatchingElement('.col');
      });
      it.each([
        ...colSizes.map(size => ['default', size, 'size', '']),
        ...colSizes.map(size => ['tablet', size, 'sm', 'sm-']),
        ...colSizes.map(size => ['tablet large', size, 'md', 'md-']),
        ...colSizes.map(size => ['desktop', size, 'lg', 'lg-']),
      ])('should show 1 column with %s breakpoint and size of %s', (breakPoint, size, breakPointType, expectedClass) => {
        const props = {
          [breakPointType]: size,
        };
        wrapper = shallow(<Col {...props}>col1</Col>);
        expect(wrapper).toContainExactlyOneMatchingElement(`.${expectedClass}col-${size}`);
      });
      it('should show multiple breakpoints', () => {
        wrapper = mount(
          <Grid>
            <Col sm="4" lg="6">col1</Col>
            <Col sm="8" lg="6">col2</Col>
          </Grid>,
        );
        expect(wrapper).toContainExactlyOneMatchingElement('BaseStyle.sm-col.sm-col-4.lg-col-6');
        expect(wrapper).toContainExactlyOneMatchingElement('BaseStyle.sm-col.sm-col-8.lg-col-6');
      });

      it('should show base breakpoint class', () => {
        wrapper = mount(
          <Grid>
            <Col md="4" lg="6">col1</Col>
            <Col md="8" lg="6">col2</Col>
          </Grid>,
        );
        expect(wrapper).toContainExactlyOneMatchingElement('BaseStyle.md-col.md-col-4.lg-col-6');
        expect(wrapper).toContainExactlyOneMatchingElement('BaseStyle.md-col.md-col-8.lg-col-6');
      });
    });
  });
});
