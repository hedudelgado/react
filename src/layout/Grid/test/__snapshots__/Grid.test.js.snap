// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component Main Container should render the component 1`] = `
<BaseStyle
  className="clearfix"
  component="div"
>
  <Col
    border={false}
  >
    test
  </Col>
</BaseStyle>
`;
